backend/testfiles/execution/stdlib/httpclient.dark

```diff
@@ -178,271 +190,354 @@ module Disallowed =
+    Builtin.Bytes.empty = PACKAGE.Darklang.Stdlib.Result.Result.Error(
+    PACKAGE.Darklang.Stdlib.HttpClient.RequestError.Other
+      "Could not connect ([::ffff:127.0.0.17]:80) Could not connect"
+  )
 
   PACKAGE.Darklang.Stdlib.HttpClient.request
     "get"
     "http://[0:0:0:0:0:ffff:7f00:0011]"
     []
-    Builtin.Bytes.empty = PACKAGE.Darklang.Stdlib.Result.Result.Error
-    "Could not connect ([::ffff:127.0.0.17]:80) Could not connect"
+    Builtin.Bytes.empty = PACKAGE.Darklang.Stdlib.Result.Result.Error(
+    PACKAGE.Darklang.Stdlib.HttpClient.RequestError.Other
+      "Could not connect ([::ffff:127.0.0.17]:80) Could not connect"
+  )
 
   PACKAGE.Darklang.Stdlib.HttpClient.request
     "get"
     "http://[0000:0000:0000:0000:0000:ffff:7f00:0011]"
     []
-    Builtin.Bytes.empty = PACKAGE.Darklang.Stdlib.Result.Result.Error
-    "Could not connect ([::ffff:127.0.0.17]:80) Could not connect"
+    Builtin.Bytes.empty = PACKAGE.Darklang.Stdlib.Result.Result.Error(
+    PACKAGE.Darklang.Stdlib.HttpClient.RequestError.Other
+      "Could not connect ([::ffff:127.0.0.17]:80) Could not connect"
+  )
 
   PACKAGE.Darklang.Stdlib.HttpClient.request
     "get"
     "http://127.255.174.17"
     []
-    Builtin.Bytes.empty = PACKAGE.Darklang.Stdlib.Result.Result.Error
-    "Bad URL: Invalid host"
+    Builtin.Bytes.empty = PACKAGE.Darklang.Stdlib.Result.Result.Error(
+    PACKAGE.Darklang.Stdlib.HttpClient.RequestError.BadUrl "Invalid host"
+  )
 
 
   PACKAGE.Darklang.Stdlib.HttpClient.request
     "get"
     "http://metadata.google.internal"
     []
-    Builtin.Bytes.empty = PACKAGE.Darklang.Stdlib.Result.Result.Error
-    "Bad URL: Invalid host"
+    Builtin.Bytes.empty = PACKAGE.Darklang.Stdlib.Result.Result.Error(
+    PACKAGE.Darklang.Stdlib.HttpClient.RequestError.BadUrl "Invalid host"
+  )
 
   PACKAGE.Darklang.Stdlib.HttpClient.request
     "get"
     "http://metadata"
     []
-    Builtin.Bytes.empty = PACKAGE.Darklang.Stdlib.Result.Result.Error
-    "Bad URL: Invalid host"
+    Builtin.Bytes.empty = PACKAGE.Darklang.Stdlib.Result.Result.Error(
+    PACKAGE.Darklang.Stdlib.HttpClient.RequestError.BadUrl "Invalid host"
+  )
 
   PACKAGE.Darklang.Stdlib.HttpClient.request
     "get"
     "http://169.254.169.254"
     []
-    Builtin.Bytes.empty = PACKAGE.Darklang.Stdlib.Result.Result.Error
-    "Bad URL: Invalid host"
+    Builtin.Bytes.empty = PACKAGE.Darklang.Stdlib.Result.Result.Error(
+    PACKAGE.Darklang.Stdlib.HttpClient.RequestError.BadUrl "Invalid host"
+  )
 
   PACKAGE.Darklang.Stdlib.HttpClient.request
     "get"
     "http://[::ffff:a9fe:a9fe]"
     []
-    Builtin.Bytes.empty = PACKAGE.Darklang.Stdlib.Result.Result.Error
-    "Could not connect ([::ffff:169.254.169.254]:80) Could not connect"
+    Builtin.Bytes.empty = PACKAGE.Darklang.Stdlib.Result.Result.Error(
+    PACKAGE.Darklang.Stdlib.HttpClient.RequestError.Other
+      "Could not connect ([::ffff:169.254.169.254]:80) Could not connect"
```

Comment:
This could be a separate case instead of an `Other`.

Actually, can you remove `Other` and make each different error a different case? You might look at the different dotnet/C# error types that can be thrown and make a different error for each one maybe.